### PR TITLE
feat: reupload

### LINKED ISSUE #313: Reupload of data
There is a new upcoming feature of Bee that allows reuploading data that is locally pinned. We should expose this.

https://github.com/ethersphere/bee/commit/8f8c05fa025db892a13a27dbee57df649955e927

### PR DESCRIPTION
**Pull request by AuHau**
Closes #313

### CODE REVIEW

File: src/modules/bzz.ts [hunk 1]
```diff
@@ -182,3 +182,19 @@ export async function uploadCollection(
...
 }
+
+/**
+ * Reupload locally pinned data
+ * @param url
+ * @param reference
+ * @param options
+ * @throws BeeResponseError if not locally pinned or invalid data
+ */
+export async function reupload(url: string, reference: Reference, options?: AxiosRequestConfig): Promise<void> {
```
Comment (agazso) [inline src/modules/bzz.ts:193]: Why there is an `options` argument? It doesn't seem to be used anywhere.

Comment (AuHau) [reply src/modules/bzz.ts:193]: Well, this was taken from the way how other modules are also implemented which allows overriding the axios configuration. I forgot to expose it in Bee class. But I am not sure if we should actually allow overriding configuration of axios like this? What is the reasoning behind it?

**Review by nugaon — COMMENTED** [review]
The code looks good, regarding to the axiosOptions on module level we can provide this additionally but I also don't see real any use-case now for that now.
for [the Bee issue](https://github.com/ethersphere/bee/issues/1803) I'd rather propose using 422 http error, as the syntax and the request are in great format and understandable by the "server", but the business-logic does not let reupload an unpinned content.

**Comment by AuHau** [comment]
> The code looks good, regarding to the axiosOptions on module level we can provide this additionally but I also don't see real any use-case now for that now.

@nugaon you mean for this reupload endpoint or generally in Bee JS?

> for the Bee issue I'd rather propose using 422 http error, as the syntax and the request are in great format and understandable by the "server", but the business-logic does not let reupload an unpinned content.

Hmm not sure how it is related to this? I would suggest to comment in the linked issue as this is up to the Bee team to decide

**Comment by nugaon** [comment]
> @nugaon you mean for this reupload endpoint or generally in Bee JS?

I meant I don't see the use-case for this reupload endpoint regarding to the axios options. anyway it can be unified as in our other modules we do the same, it doesn't pollute the code.

**Comment by nugaon** [comment]
> Hmm not sure how it is related to this? I would suggest to comment in the linked issue as this is up to the Bee team to decide

it was commented in your code in this PR.

**Pull request merged**